talk/owt/sdk/p2p/p2ppeerconnectionchannel.cc

```diff
@@ -761,6 +761,11 @@ void P2PPeerConnectionChannel::OnSetLocalSessionDescriptionSuccess() {
   // Setting maximum bandwidth here.
   ApplyBitrateSettings();
   auto desc = LocalDescription();
+  if (!desc) {
```

Comment:
the problem is a race with `ClosePeerConnection()`. The problem is line 757-759 checks `ended_` under a lock, but once that lock is released, all bets are off. 

The solution we've used is the `GetPeerConnectionRef()` method. It internally checks the `ended_` flag and returns a reference counted version of the peer connection. That will guarantee the peer_connection_ doesn't become null before the method exits. You can simply replace line 757-760 with this method call and no other changes needed

Reply:
I actually realized this as I was looking at [this bug](https://ambient-ai.atlassian.net/browse/AMB-16882), but you beat me to it. :-D

Reply:
Also, we still need to account for the peer connection ref being null, so I'll move the block from 757-772 under the if check for the ptr.

Reply:
This seems to be a common problem though - I see the peer connection isn't checked elsewhere either. Is that intentional @davidtag? 

`  rtc::scoped_refptr<webrtc::PeerConnectionInterface> temp_pc_ = GetPeerConnectionRef();
`
`temp_pc_` could be null, but the caller assumes it's not.

Reply:
not intentional, likely bugs